### PR TITLE
Fix issue 19986 - Can't assign large const T to std.Variant.VariantN

### PR DESCRIPTION
**Pull request by shove70**

### CODE REVIEW

**Comment by John-Colvin** [comment]
A solution should probably handle this case too:

```D
import std.variant;
void foo()
{
    VariantN!32 v;
    v = const(S).init;
}

struct S
{
    ubyte[33] s;
}
```

**Comment by shove70** [comment]
> A solution should probably handle this case too:
> 
> ```d
> import std.variant;
> void foo()
> {
>     VariantN!32 v;
>     v = const(S).init;
> }
> 
> struct S
> {
>     ubyte[33] s;
> }
> ```

Okay, let me deal with it. Thank you for reminding me.

**Comment by shove70** [comment]
@John-Colvin, Help me see, is it feasible to solve this issue? Thank you!

File: std/variant.d [hunk 1]
```diff
@@ -690,12 +690,29 @@ public:
...
-                    *p = rhs;
+                    static if (!isMutable!T)
+                    {
+                        alias UT = Unqual!T;
+                        auto p = cast(UT*)(new U[n]).ptr;
+                        *p = cast(UT) rhs;
+                    }
+                    else
+                    {
+                        auto p = cast(T*)(new U[n]).ptr;
```
Comment (thewilsonator) [inline std/variant.d]: You can just drop this branch and use `Unqual` everywhere since `Unqual!S == S`

Comment (shove70) [reply std/variant.d]: Thanks!
It has been handled as review required.

**Pull request merged**